simulator/README.md

```diff
@@ -0,0 +1,52 @@
+* Supporting different message replay strategies.
+* Maintaining persistant state across restarts and failures.
+
+You can run the simulator with:
+```
+go build
+./simulator --messages=messages.mllp --akis=akis.csv --groups=groups.yaml
+```
+* `--messages` specifies the file containing the messages to replay. The messages should be be encapsulated within MLLP, which could be generated by the [generator](../generator/README.md) in this repo.
+* `--akis` specifies the file containing the patient blood test events that represent a diagnosis of acute kidney injury, which could be generated by the [generator](../generator/README.md) in this repo.
+* `--groups` specifies a YAML configuration file detailing the independent groups the simulator should serve on different ports. The format of the configuration file is described below. Without this flag, the simulator serves a single group.
+* `--mllp` specifies the port on which to listen for MLLP connections, if `--groups` is not used and metrics for only a single group are tracked.
+* `--pager` specifies the the port on which to listen for pager request, over HTTP, if `--groups` is not used and metrics for only a single group are tracked.
+* `--addr` the address on which to serve MLLP and HTTP, for all ports. Defaults to `0.0.0.0`.
+* `--backstage` the port on which to serve the simple internal status UI.
+* `--state` the directory in which the simulator should persist it's state, as plain text log files, one per group.
+* `--simulated-time` an arbitrary simulated time, specified in HL7 format (eg `20240105`). If not specifed, defaults to 7am in the morning on the first day with a blood test result reported in `messages.mllp`.
+* `--real-time` the real time corresponding to the time passed as `--simulated-time`, specified in HL7 format (eg `20240308`)
+* `--replay-in-real-time=true` replay messages in real time, using the times specified by `--simulated-time` and `--real-time` to determine when a given message should be sent. If not set, messages are sent as quickly as clients can acknowledge them.
+* `--rewind=false` only send a given HL7 message to a group once, persistantly tracking the last message sent to each group. If not set, messages are replayed from the start of `messages.mllp` each time a group connects.
+
+### Configuration file format
+The YAML configuration file should look like this:
+```
+groups:
+- name: mimico
+  mllp: 9440
+  pager: 10440
+- name: volantin
+  mllp: 9441
+  pager: 10441
+
+```
+In this case, the simulator is serving two different groups, named `mimico` and `volantin`, using ports 9440 and 9441 to listen for MLLP connections from the respective groups, and 10440 and 10441 for pager requests. Using group names inspired by the [bridges of architect Santiago Calatrava](https://calatrava.com/projects.html?project_type=bridges) is encouraged, but not required.
```

Comment:
cute bridges

Reply:
Aren't they?? My favourite is [Serreria](https://calatrava.com/projects/serreria-bridge-valencia.html).